src/Engines/CollectionEngine.php

```diff
@@ -0,0 +1,239 @@
+            return $models;
+        }
+
+        $columns = array_keys($models->first()->toSearchableArray());
+
+        return $models->filter(function ($model) use ($builder, $columns) {
+            foreach ($columns as $column) {
+                $attribute = $model->{$column};
+
+                if (Str::contains($attribute, $builder->query)) {
+                    return true;
+                }
+            }
+
+            return false;
+        })->values();
+    }
+
+    /**
+     * Pluck and return the primary keys of the given results.
+     *
+     * @param  mixed  $results
+     * @return \Illuminate\Support\Collection
+     */
+    public function mapIds($results)
+    {
+        $results = $results['results'];
+
+        return count($results) > 0
+                    ? collect($results)->pluck($results[0]->getKeyName())->values()
+                    : collect();
+    }
+
+    /**
+     * Map the given results to instances of the given model.
+     *
+     * @param  \Laravel\Scout\Builder  $builder
+     * @param  mixed  $results
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @return \Illuminate\Database\Eloquent\Collection
+     */
+    public function map(Builder $builder, $results, $model)
+    {
+        $results = $results['results'];
+
+        if (count($results) === 0) {
+            return $model->newCollection();
+        }
+
+        $objectIds = collect($results)
+                ->pluck($model->getKeyName())
+                ->values()
+                ->all();
+
+        $objectIdPositions = array_flip($objectIds);
+
+        return $model->getScoutModelsByIds(
+            $builder, $objectIds
+        )->filter(function ($model) use ($objectIds) {
+            return in_array($model->getScoutKey(), $objectIds);
+        })->sortBy(function ($model) use ($objectIdPositions) {
+            return $objectIdPositions[$model->getScoutKey()];
+        })->values();
+    }
+
+    /**
+     * Map the given results to instances of the given model via a lazy collection.
+     *
+     * @param  \Laravel\Scout\Builder  $builder
+     * @param  mixed  $results
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @return \Illuminate\Support\LazyCollection
+     */
+    public function lazyMap(Builder $builder, $results, $model)
+    {
+        $results = $results['results'];
+
+        if (count($results) === 0) {
+            return LazyCollection::make($model->newCollection());
+        }
+
+        $objectIds = collect($results)
+                ->pluck($model->getKeyName())
+                ->values()->all();
+
+        $objectIdPositions = array_flip($objectIds);
+
+        return $model->queryScoutModelsByIds(
+                $builder, $objectIds
+            )->cursor()->filter(function ($model) use ($objectIds) {
+                return in_array($model->getScoutKey(), $objectIds);
+            })->sortBy(function ($model) use ($objectIdPositions) {
```

Comment:
Sort, by definition, cannot be lazy.

This means that as soon as a single item is pulled out at the end of the chain, it'll have to pull all records into memory.

Reply:
So should I just switch back to `get` instead of `cursor`? I don't have big memory concerns here. This driver is only for local development.

Reply:
In this scenario, there isn't much practical difference between the two.